theano/tensor/nlinalg.py

```diff
@@ -726,3 +726,37 @@ def norm(x, ord):
             raise ValueError(0)
     elif ndim > 2:
         raise NotImplementedError("We don't support norm witn ndim > 2")
+
+
+class LogAbsDet(Op):
+    """Computes the logarithm of absolute determinant of a square
+    matrix M, log(abs(det(M))), on CPU. Avoids det(M) overflow/
+    underflow.
+
+    TODO: add GPU code!
+    """
+    def make_node(self, x):
+        x = theano.tensor.as_tensor_variable(x)
+        o = theano.tensor.scalar(dtype=x.dtype)
+        return Apply(self, [x], [o])
+
+    def perform(self, node, inputs, outputs):
+        try:
+            (x,) = inputs
+            (z,) = outputs
+            s = numpy.linalg.svd(x, compute_uv=False)
+            log_abs_det = numpy.sum(numpy.log(numpy.abs(s)))
```

Comment:
I didn't check the math. But this can be done by reusing current op. The svd op and the elemwise op. This will be better as the elemwise op will get optimized by Theano and will work on the GPU (for the elemwise).

Then the only think missing to make it work on the GPU would be to have an svn op on the GPU. There is one implementation here that won't request c code: http://scikit-cuda.readthedocs.org/en/latest/generated/skcuda.linalg.svd.html